components/nexusoperations/executors.go

```diff
@@ -566,6 +575,37 @@ func (e taskExecutor) executeCancelationBackoffTask(env hsm.Environment, node *h
 	})
 }
 
+func (e taskExecutor) extractEndpointNameFromOperation(ctx context.Context, env hsm.Environment, ref hsm.Ref, getOpNode func(*hsm.Node) *hsm.Node) (endpoint string, err error) {
+	err = env.Access(ctx, ref, hsm.AccessRead, func(n *hsm.Node) error {
+		op, err := hsm.MachineData[Operation](getOpNode(n))
+		if err != nil {
+			return err
+		}
+		endpoint = op.Endpoint
+		return nil
+	})
+	return
+}
+
+// lookupEndpint gets an endpoint from the registry, preferring to look up by ID and falling back to name lookup.
+// The fallback is a temporary workaround for not implementing endpoint replication, and endpoint ID being a UUID set by
+// the system. We try to get the endpoint by name to support cases where an operator manually created an endpoint with
+// the same name in two replicas.
+func (e taskExecutor) lookupEndpoint(ctx context.Context, namespaceID namespace.ID, endpointID, endpointName string) (*persistencespb.NexusEndpointEntry, error) {
+	entry, err := e.EndpointRegistry.GetByID(ctx, endpointID)
+	if err != nil {
+		if errors.As(err, new(*serviceerror.NotFound)) {
+			entry, err := e.EndpointRegistry.GetByName(ctx, namespaceID, endpointName)
```

Comment:
nit: you could just directly return the result of `GetByName`

Reply:
Thanks!